cypress/e2e/requests.cy.js

```diff
@@ -2,50 +2,102 @@ describe('Viewing all requests', () => {
   describe('as a logged out user', () => {
     it('should show an error message.', () => {
       // Visit a protected route in order to allow cypress to set the cookie and mock the login
-      cy.visit("/requests")
+      cy.visit('/requests')
       cy.get('div.alert-heading').contains('Unauthorized').then(() => {
-        cy.log("A logged out user is not able to view requests.")
+        cy.log('A logged out user is not able to view requests.')
       })
     })
   })
   
   describe('as a logged in user', () => {
     let scientistApiBaseURL = `https://${Cypress.env('NEXT_PUBLIC_PROVIDER_NAME')}.scientist.com/api/v2`
+    // declare variables that can be used to change how the response is intercepted.
     let requestList
+    let loading
+    let error
 
     beforeEach(() => {
       // Call the custom cypress command to log in
       cy.login(Cypress.env('TEST_SCIENTIST_USER'), Cypress.env('TEST_SCIENTIST_PW'))
-
       // Intercept the response from the endpoint to view all requests
+      // TODO(summer-cook): extract out this base url into the config to use as an environment variable. it was not cooperating before
       cy.intercept('GET', `${scientistApiBaseURL}/quote_groups/mine.json`, (req) => {
-        if (requestList === true) {
-          req.reply({fixture: 'all-requests/requests.json'})
-        } else {
-          req.reply({fixture: 'all-requests/no-requests.json'})
+        if ((requestList === undefined) && (loading === true)) {
+          // reply with an empty response: both data and error will be undefined.
+          req.reply()
+        } else if ((requestList === undefined) && (loading === false) && (error === true)) {
+          // error will be defined
+          req.reply({ statusCode: 404 })
+        } else if (requestList === true) {
+          // reply with a request body- default status code is 200
+          req.reply({ fixture: 'all-requests/requests.json' })
+        } else if (requestList === false) {
```

Comment:
this (and the corresponding change in the test to set requestList = []) causes the test to fail for some reason which doesnt really make sense to me since the case above for true uses strict equals

Reply:
i also tried to set requestList in the tests to the actual fixture that is being used but that doesn't work either since the fixture is an object with quote_group_refs inside, which is truthy